src/api/icon.api.js

```diff
@@ -1,42 +1,142 @@
+     *   belong to one icon set)
+     * @param {Number[]} anchor Offset to apply to this icon when placed on a coordinate ([x,y] format)
+     */
+    constructor(name, imageURL, imageTemplateURL, iconSetName, anchor) {
+        this._name = name
+        this._imageURL = imageURL
+        this._imageTemplateURL = imageTemplateURL
+        this._iconSetName = iconSetName
+        this._anchor = anchor
+    }
+
+    /** @returns {String} Name of this icon in the backend (lower cased) */
+    get name() {
+        return this._name
+    }
+
+    /** @returns {String} URL to the image of this icon itself (with default size and color) */
+    get imageURL() {
+        return this._imageURL
+    }
+
+    /**
+     * @returns {String} URL template where size and color can be defined (by replacing {icon_scale}
+     *   and {{r},{g},{b}} respectively, see {@link Icon.generateURL})
+     */
+    get imageTemplateURL() {
+        return this._imageTemplateURL
+    }
+
+    /** @returns {Number[]} Offset to apply to this icon when placed on a coordinate ([x,y] format) */
+    get anchor() {
+        return this._anchor
+    }
+
+    /**
+     * @returns {String} Name of the {@link IconSet} in which belongs this icon (an icon can only
+     *   belong to one icon set)
+     */
+    get iconSetName() {
+        return this._iconSetName
     }
 
     /**
-     * Generate an icon URL from its template
+     * Generate an icon URL from its template. If no iconSize is given, medium scale will be
+     * applied. If no iconColor is given, red will be applied (if applicable, as non-colorable icons
+     * will not have {r}, {g}, {b} part of their template URL)
      *
-     * @param {String} iconSetName Name of the icon set, should come from an {IconSet} directly
-     * @param {DrawingStyleSize} iconSize The size (or scale) to use for this icon's URL
-     * @param {DrawingStyleColor} iconColor The color to use for this icon's URL
+     * @param {FeatureStyleSize} iconSize The size (or scale) to use for this icon's URL
+     * @param {FeatureStyleColor} iconColor The color to use for this icon's URL
      * @returns {String} A full URL to this icon on the service-icons backend
      */
-    generateURL(iconSetName, iconSize, iconColor) {
+    generateURL(iconSize = MEDIUM, iconColor = RED) {
         const rgb = iconColor.rgb.slice(0, 3)
-        return this.imageTemplateURL
-            .replace('{icon_set_name}', iconSetName)
-            .replace('{icon_name}', this.name)
+        return this._imageTemplateURL
+            .replace('{icon_set_name}', this._iconSetName)
+            .replace('{icon_name}', this._name)
             .replace('{icon_scale}', iconSize.iconScale + 'x')
-            .replace('{r}', rgb[0])
-            .replace('{g}', rgb[1])
-            .replace('{b}', rgb[2])
+            .replace('{r}', `${rgb[0]}`)
+            .replace('{g}', `${rgb[1]}`)
+            .replace('{b}', `${rgb[2]}`)
```

Comment:
I'm curious. Did this pose a problem? (when passed directly)

Reply:
My IDE was complaining about the unmatching types (JS will do the conversion behind the scene anyway...)
I don't remember if the linter was also complaining though

Reply:
Ok :smiley: